parfait-core/src/main/java/io/pcp/parfait/timing/MetricMeasurement.java

```diff
@@ -38,8 +44,14 @@
     private final Thread thread;
 
     public MetricMeasurement(ThreadMetric metricSource, Thread thread) {
-        this.metricSource = metricSource;
+        this.metricSource = metricSource;        
         this.thread = thread;
+        if (metricSource != null && LOG.isWarnEnabled()) {
```

Comment:
Why is this logged at a WARN level?  What "Warning" is this log statement indicating?  It's more a DEBUG level log by the looks of it?